bundles/org.openhab.binding.luxom/src/main/java/org/openhab/binding/luxom/internal/handler/LuxomBridgeHandler.java

```diff
@@ -0,0 +1,354 @@
+        this.communication = new LuxomCommunication(this);
+    }
+
+    @Override
+    public void handleCommand(@NotNull ChannelUID channelUID, Command command) {
+        logger.debug("bridge received command {} for {}", command.toFullString(), channelUID);
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getThing().getConfiguration().as(LuxomBridgeConfig.class);
+
+        if (validConfiguration(this.config)) {
+            reconnectInterval = (config.reconnectInterval > 0) ? config.reconnectInterval
+                    : DEFAULT_RECONNECT_INTERVAL_IN_MINUTES;
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                    localizationService.getText("status.connecting", "Connecting"));
+            scheduler.submit(this::connect); // start the async connect task
+        }
+    }
+
+    private boolean validConfiguration(@Nullable LuxomBridgeConfig config) {
+        if (config == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    localizationService.getText("status.bridge-configuration-missing", "bridge configuration missing"));
+
+            return false;
+        }
+
+        if (config.ipAddress == null || config.ipAddress.trim().isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    localizationService.getText("status.bridge-address-missing", "bridge address not specified"));
+
+            return false;
+        }
+
+        return true;
+    }
+
+    private void scheduleConnectRetry(long waitMinutes) {
+        logger.warn("Scheduling connection retry in {} (minutes)", waitMinutes);
+        connectRetryJob = scheduler.schedule(this::connect, waitMinutes, TimeUnit.MINUTES);
+    }
+
+    private synchronized void connect() {
+        if (this.communication.isConnected()) {
+            return;
+        }
+
+        if (config != null) {
+            logger.debug("Connecting to bridge at {}", config.ipAddress);
+        }
+
+        try {
+            communication.startCommunication();
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            disconnect();
+            scheduleConnectRetry(reconnectInterval); // Possibly a temporary problem. Try again later.
+        }
+    }
+
+    public void startProcessing() {
+        this.nrOfSendPermits.set(1);
+
+        updateStatus(ThingStatus.ONLINE);
+
+        messageSender = new Thread(this::sendCommandsThread, "Luxom sender");
+        messageSender.start();
+
+        logger.debug("Starting heartbeat job with interval {} (seconds)", HEARTBEAT_INTERVAL_SECONDS);
+        heartBeat = scheduler.scheduleWithFixedDelay(this::sendHeartBeat, 10, HEARTBEAT_INTERVAL_SECONDS,
+                TimeUnit.SECONDS);
+    }
+
+    private void sendCommandsThread() {
+        logger.info("Starting send commands thread...");
```

Comment:
Definitely should not be info, but can probably be omitted entirely. See https://www.openhab.org/docs/developer/guidelines.html#f-logging